homeassistant/components/atlona_juno/media_player.py

```diff
@@ -0,0 +1,156 @@
+"""Support for interfacing with Atlona Juno 451 HDMI 4x1 Switch."""
+import logging
+
+from pyatlonajuno.lib import Juno451
+import voluptuous as vol
+
+from homeassistant.components.media_player import MediaPlayerDevice, PLATFORM_SCHEMA
+from homeassistant.components.media_player.const import (
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_PROTOCOL,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_NAME,
+    STATE_OFF,
+    STATE_ON,
+)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Atlona Juno 451"
+DEFAULT_PORT = 80
+DEFAULT_PROTOCOL = "http"
+
+DATA_ATLONAJUNO = "atlona_juno"
+
+SUPPORT_ATLONAJUNO = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+CONF_SOURCES = "sources"
+ATTR_SOURCE = "source"
+
+# Valid source ids: 1-4
+MEDIA_PLAYER_SCHEMA = vol.Schema({ATTR_ENTITY_ID: cv.comp_entity_ids})
+SOURCE_IDS = vol.All(vol.Coerce(int), vol.Range(min=1, max=4))
+SOURCE_SCHEMA = vol.Schema({vol.Required(CONF_NAME): cv.string})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PROTOCOL, default=DEFAULT_PROTOCOL): cv.string,
+        vol.Required(CONF_SOURCES): vol.Schema({SOURCE_IDS: SOURCE_SCHEMA}),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Atlona Juno 451 HDMI switch."""
+    if DATA_ATLONAJUNO not in hass.data:
+        hass.data[DATA_ATLONAJUNO] = {}
+
+    url = "{}://{}:{}".format(
+        config.get(CONF_PROTOCOL), config.get(CONF_HOST), config.get(CONF_PORT)
+    )
+    _LOGGER.debug("Composed URL for Atlona Juno %s", url)
+    sources = {
+        source_id: extra[CONF_NAME] for source_id, extra in config[CONF_SOURCES].items()
+    }
+
+    # As a device has one and only one IP address in our context, this qualiffies for uid
+    unique_id = "{}-{}".format(DATA_ATLONAJUNO, config.get(CONF_HOST).replace(".", "_"))
+    atlona_device = AtlonaJuno(config.get(CONF_NAME), unique_id, Juno451(url), sources)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.